image_transport/include/image_transport/subscriber_plugin.hpp

```diff
@@ -147,11 +200,12 @@ class IMAGE_TRANSPORT_PUBLIC SubscriberPlugin
    * \brief Subscribe to an image transport topic. Must be implemented by the subclass.
    */
   virtual void subscribeImpl(
-    rclcpp::Node * node,
     const std::string & base_topic,
     const Callback & callback,
     rmw_qos_profile_t custom_qos,
     rclcpp::SubscriptionOptions options) = 0;
+
+  std::shared_ptr<NodeType> node_;
```

Comment:
by holding a reference to the Node here, are we going to end up with some sort of hanging reference where the node never goes out of scope? (I don't recall the specifics, but this was a big problem for Nav2 and they eventually switched to storing WeakPtrs inside the various plugins so that the SharedPtr would eventually go out of scope as expected)

Reply:
Fixed as these are not needed anymore.